### PR TITLE
Added prerequisites script

### PR DESCRIPTION
**Pull request by e-minguez**

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/297/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/303/

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,11 @@
 .PHONY: default all requirements configure repo_sync ironic build ocp_run clean ocp_cleanup host_cleanup
-default: requirements configure repo_sync ironic build ocp_run
+default: prereq requirements configure repo_sync ironic build ocp_run
 
 all: default
 
+prereq:
+	./00_prerrequisites.sh
```
Comment (derekhiggins) [inline Makefile]: CI failed because of a extra "r" here

File: Makefile [hunk 2]
```diff
@@ -22,7 +25,7 @@ ocp_run:
 	./06_deploy_bootstrap_vm.sh
 	./07_deploy_masters.sh
 
-clean: masters_cleanup ocp_cleanup host_cleanup
+clean: prereq masters_cleanup ocp_cleanup host_cleanup
```
Comment (derekhiggins) [inline Makefile]: Why do we need to add it to clean?

File: Makefile (hunk 1 above)
Comment (e-minguez) [reply Makefile]: My bad, fixed. In Spanish we double 'r' if surrounded by vowels depending on the sound (like in 'perro').

File: Makefile (hunk 2 above)
Comment (e-minguez) [reply Makefile]: Just in case. It doesn't hurt, but I'll remove it if needed.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/306/

**Comment by e-minguez** [comment]
`lsb_release: command not found`

TIL about http://0pointer.de/blog/projects/os-release.html

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/307/

**Comment by e-minguez** [comment]
> Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/307/

`Build timed out (after 60 minutes). Marking the build as failed.`

File: Makefile [hunk 3]
```diff
@@ -1,8 +1,11 @@
 .PHONY: default all requirements configure repo_sync ironic build ocp_run clean ocp_cleanup host_cleanup
```
Comment (russellb) [inline Makefile]: need to add `prereq` here

File: Makefile (hunk 2 above)
Comment (hardys) [reply Makefile]: Yeah we don't need to check prerequisites in clean IMHO, it's a pre-deploy check

File: Makefile (hunk 3 above)
Comment (hardys) [reply Makefile]: Yeah but then we'd run them multiple times when common.sh is sourced in multiple scripts?

If we do that we should probably drop a lock file in the `ocp` dir so we can run only once after `make clean`

Comment (hardys) [reply Makefile]: Oops, this is referring to the previous `common.sh` comment

File: Makefile (hunk 2 above)
Comment (e-minguez) [reply Makefile]: I'll remove it, thanks!

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/321/

**Comment by e-minguez** [comment]
I've redid the PR to be integrated in the common.sh as there were some checks in that scripts. Now the common.sh:

* Gets the config_${user} file
* Set all variables
* Checks $stuff
* Do things (create the working dir if not available and the log stuff)

Let me know if it is better now :)

Thanks!

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/322/

**Comment by e-minguez** [comment]
I've redid the PR to be integrated in the common.sh as there were some checks in that script. Now the common.sh:

    Gets the config_${user} file
    Set all variables
    Checks $stuff
    Do things (create the working dir if not available and the log stuff)

Let me know if it is better now :)

Thanks!

File: Makefile (hunk 3 above)
Comment (e-minguez) [reply Makefile]: I've added the prereq to the common.sh as there were some other prerequisites there.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/323/

**Comment by e-minguez** [comment]
Is there anything else required for this? /cc @russellb 
Thanks!

File: common.sh [hunk 4]
```diff
@@ -77,6 +57,73 @@ export IRONIC_DATA_DIR="$WORKING_DIR/ironic"
...
 export KUBECONFIG="${SCRIPTDIR}/ocp/auth/kubeconfig"
 
+# Use a cloudy ssh that doesn't do Host Key checking
+export SSH="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5"
+
+# Verify requisites/permissions
+# Connect to system libvirt
+export LIBVIRT_DEFAULT_URI=qemu:///system
+if [ "$USER" != "root" -a "${XDG_RUNTIME_DIR:-}" == "/run/user/0" ] ; then
+    echo "Please use a non-root user, WITH a login shell (e.g. su - USER)"
```
Comment (vrutkovs) [inline common.sh]: Why though?

Comment (e-minguez) [reply common.sh]: This wasn't added in this PR tho (https://github.com/openshift-metalkube/dev-scripts/blob/2f4884a79e669a7b90655aecbaa0d60ce4b820fb/common.sh#L34-L38)

**Pull request merged**